src/richtext.ts

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import StoryblokComponent from './common/storyblok-component';
+import type { StoryblokRichTextNode, StoryblokRichTextOptions } from '@storyblok/js';
+import { BlockTypes, richTextResolver } from '@storyblok/js';
+
+function componentResolver(node: StoryblokRichTextNode<React.ReactElement>) {
+  // Convert this to use React.createElement or JSX
+  // Example with JSX:
+  return React.createElement(StoryblokComponent, {
+    blok: node?.attrs?.body[0],
```

Comment:
I would double check on `body` before accessing element `0` to avoid runtime errors, also it would be great to add a simple unit test to check this function with various test inputs as `node` to be sure it does'n break

Reply:
Mmm, `componentResolver` is meant to be only internal, it's not something users should use and the API should always ensure to return a body for the component on the API. Still, I added some unit tests as you requested.